vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -0,0 +1,343 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+
+package VASSAL.build.module.map;
+
+import java.awt.event.MouseEvent;
+import java.awt.event.ActionEvent;
+import VASSAL.build.module.documentation.HelpFile;
+import VASSAL.command.Command;
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Graphics;
+import VASSAL.build.GameModule;
+import VASSAL.build.Buildable;
+import javax.swing.Timer;
+
+import org.apache.commons.lang3.SystemUtils;
+
+import java.awt.Point;
+import VASSAL.build.module.Map;
+import java.awt.event.MouseListener;
+import java.awt.event.ActionListener;
+import VASSAL.build.module.GameComponent;
+import VASSAL.build.module.GlobalOptions;
+import VASSAL.command.CommandEncoder;
+import VASSAL.build.AbstractConfigurable;
+import VASSAL.build.AutoConfigurable;
+import VASSAL.command.FlareCommand;
+import VASSAL.configure.ColorConfigurer;
+import VASSAL.configure.StringEnum;
+import VASSAL.i18n.Resources;
+import VASSAL.tools.swing.SwingUtils;
+
+public class Flare extends AbstractConfigurable
+    implements CommandEncoder, GameComponent, Drawable, ActionListener, MouseListener {
+  public static final String COMMAND_PREFIX = "FLARE:";
+  private Map map;
+  private int circleSize;
+  private int pulses;
+  private int pulsesPerSec;
+  private int frame;
+  private boolean animate;
+  private int frames;
+  private int framesPerPulse;
+  private String flareKey;
+  private Color color;
+  private static int CIRCLE_RATE    = 33; // 33ms for approx 30 frames/sec
+  private static int DEFAULT_FRAMES = 60; // 2 sec for basic "no animation" pulse
+  private Point clickPoint;
+  private Boolean active;
+  private Timer timer;
+  public static final String CIRCLE_SIZE  = "circleSize";
+  public static final String CIRCLE_COLOR = "circleColor";
+  public static final String FLARE_KEY    = "flareKey";
+  public static final String PULSES       = "flarePulses";
+  public static final String PULSES_PER_SEC = "flarePulsesPerSec";
+  
+  public static final String FLARE_ALT_LOCAL     = Resources.getString("Editor.Flare.flare_key_desc", Resources.getString("Keys.alt"));
+  public static final String FLARE_CTRL_LOCAL    = Resources.getString("Editor.Flare.flare_key_desc", Resources.getString("Keys.ctrl"));
+  public static final String FLARE_COMMAND_LOCAL = Resources.getString("Editor.Flare.flare_key_desc", Resources.getString("Keys.meta"));
+  
+  public static final String FLARE_ALT = "keyAlt";
+  public static final String FLARE_CTRL = "keyCtrl";
+  
+
+
+  public Flare() {
+    this.circleSize = 100;
+    this.color = Color.RED;
+    this.active = false;
+    this.flareKey = FLARE_ALT;
+    this.pulses       = 6;
+    this.pulsesPerSec = 3;
```

Comment:
I think every `this` in the whole file is superfluous. (You generally only need it in places where there's potential for ambiguity.)